diff --git a/package/nodejs/nodejs-src/nodejs-src.mk b/package/nodejs/nodejs-src/nodejs-src.mk
index 8f934202b0..9974808c26 100644
--- a/package/nodejs/nodejs-src/nodejs-src.mk
+++ b/package/nodejs/nodejs-src/nodejs-src.mk
@@ -21,8 +21,6 @@ NODEJS_SRC_DEPENDENCIES = \
 	host-pkgconf \
 	host-python3 \
 	host-qemu \
-	c-ares \
-	libuv \
 	nghttp2 \
 	zlib \
 	$(call qstrip,$(BR2_PACKAGE_NODEJS_MODULES_ADDITIONAL_DEPS))
@@ -37,8 +35,6 @@ NODEJS_SRC_INSTALL_STAGING = YES
 
 NODEJS_SRC_CONF_OPTS = \
 	--shared-zlib \
-	--shared-cares \
-	--shared-libuv \
 	--shared-nghttp2 \
 	--cross-compiling \
 	--dest-os=linux \
diff --git a/package/nodejs/nodejs.hash b/package/nodejs/nodejs.hash
index 486e5fce9a..29c96c528b 100644
--- a/package/nodejs/nodejs.hash
+++ b/package/nodejs/nodejs.hash
@@ -1,10 +1,9 @@
-# From https://nodejs.org/dist/v20.18.2/SHASUMS256.txt.asc
-sha256  05819d72dcc0aa788baab1066e18ede5f1ab6730a1925cd6b15c131b55fd4272  node-v20.18.2-headers.tar.xz
-sha256  5c1437aa16e7e6a2e0687a42c4d3f0a8f8a2039cda8880cb3be8cd983aeefb44  node-v20.18.2-linux-arm64.tar.xz
-sha256  63d4df56fb2e34a5077345f78941094204d2223ce03b8ebc9c1500e6e2aae68d  node-v20.18.2-linux-armv7l.tar.xz
-sha256  828a2635261ca225cd4a8a4b1a914003cdc7b30656c2e9092ac7aab02ac361db  node-v20.18.2-linux-ppc64le.tar.xz
-sha256  4e50f727ae09bdafecf2322c72faf7cd82bf3b8851a16b8bb63974e0d8d6eceb  node-v20.18.2-linux-x64.tar.xz
-sha256  69bf81b70f3a95ae0763459f02860c282d7e3a47567c8afaf126cc778176a882  node-v20.18.2.tar.xz
+# From https://nodejs.org/dist/v24.18.0/SHASUMS256.txt.asc
+sha256  694c4868c187d4a435a86d617346e2c81676c30a9bbf6fc65cce8eb50038dcaf  node-v24.18.0-headers.tar.xz
+sha256  58c9520501f6ae2b52d5b210444e24b9d0c029a58c5011b797bc1fe7105886f6  node-v24.18.0-linux-arm64.tar.xz
+sha256  9717e491a75590c304d17fc191179c9613907b0fddfce22106995d612f23953f  node-v24.18.0-linux-ppc64le.tar.xz
+sha256  55aa7153f9d88f28d765fcdad5ae6945b5c0f98a36881703817e4c450fa76742  node-v24.18.0-linux-x64.tar.xz
+sha256  e94afde24db08e0c564ee7110a2d5aab51ee0059382c9fd8233c54eec47b28f9  node-v24.18.0.tar.xz
 
 # Locally calculated
-sha256  4f0e8660d7fe2f8f7759a54c53907300c555992b580e846b30b75121c53ad180  LICENSE
+sha256  148eacf7863ef4329224a29398623077200a27194aa075569faf4a0a85566ca5  LICENSE
diff --git a/package/nodejs/nodejs.mk b/package/nodejs/nodejs.mk
index 2bdebcd853..877420d338 100644
--- a/package/nodejs/nodejs.mk
+++ b/package/nodejs/nodejs.mk
@@ -5,7 +5,7 @@
 ################################################################################
 
 # _VERSION, _SOURCE and _SITE must be kept empty to avoid downloading anything
-NODEJS_COMMON_VERSION = 20.18.2
+NODEJS_COMMON_VERSION = 24.18.0
 NODEJS_COMMON_SOURCE = node-v$(NODEJS_COMMON_VERSION).tar.xz
 NODEJS_COMMON_SITE = http://nodejs.org/dist/v$(NODEJS_COMMON_VERSION)
 
